packages/terra-button/src/Button.jsx

```diff
@@ -229,10 +217,9 @@ class Button extends React.Component {
 
     const ariaLabel = customProps['aria-label'];
 
-    const ComponentType = href ? 'a' : 'button';
-
+    /* eslint-disable react/button-has-type */
```

Comment:
Rather than disabling the lint rule here, you could default `type="button"` and then allow the type to be overridden if the prop is passed in. I think that would be passive.

Reply:
"type" attribute for button doesn't seem to support dynamic assignment (https://github.com/jsx-eslint/eslint-plugin-react/issues/1555), couldn't make use of conditionals